kibana/templates/deployment.yaml

```
@@ -37,6 +37,8 @@ spec:
 {{ toYaml .Values.podSecurityContext | indent 8 }}
       {{- if .Values.serviceAccount }}
       serviceAccount: {{ .Values.serviceAccount }}
+      {{- else if .Values.managedServiceAccount }}
```

I think this one needs to be looking at the `.create` key instead.
```suggestion
      {{- else if .Values.managedServiceAccount.create }}
```